code-challenges.test.js

```diff
@@ -14,44 +16,81 @@
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
 
-// b) Create the function that makes the test pass.
+  return array;
+}
 
 // Pseudo code:
+// the functions name is shuffle (array) the argument will be an array
+// us the shift method to to take of  the element at 0 index
+// use a for loop to loop throught the array and then shuffle it
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
-
+describe("voteCount", () => {
+  it("takes in an object that contains up votes and down votes and returns the end tally.", () => {
+    const votes1 = { upVotes: 13, downVotes: 2 }
+    const votes2 = { upVotes: 2, downVotes: 33 }
+    expect(voteCount(votes1)).toEqual(11)
+    expect(voteCount(votes2)).toEqual(-31)
+  })
+})
+// b) Create the function that makes the test pass.
 const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
 
-// b) Create the function that makes the test pass.
+const voteCount = (object) => {
+return object.upVotes - object.downVotes
+}
 
 // Pseudo code:
+// Create function named voteCount it will one parameter that will accept and obeject
+// on the passed in object access the first and second key
+// return the result of subtracting the value of the first and second key
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
-
-const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
-
+describe("noDuplicates", () => {
+  it("takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.", () => {
+    const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    expect(noDuplicates(dataTypesArray1,dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 // b) Create the function that makes the test pass.
+const noDuplicates = (array1, array2) => {
+  let newArray = array1.concat([...array2])
+  let uniqueArray = [...new Set(newArray)];
+  return uniqueArray
+}
 
 // Pseudo code:
+// named the function noDuplicates 
+// takes two arrays as argument
+// concat the two arrays into one new variable
+// use some sort of filter to get the duplicates off
```

Comment:
This pseudo code does not provide an explanation to your code. Instead of saying "use some sort of filter", your pseudo code should be actually explaining what your code is doing. So you should have explained utilizing a Set to remove duplicates, and putting the values of the Set into an array using the spread operator.